src/Components/Components/src/Rendering/RenderTreeBuilder.cs

```diff
@@ -682,10 +682,8 @@ public void AddComponentReferenceCapture(int sequence, Action<object> componentR
     /// Adds a frame indicating the render mode on the enclosing component frame.
     /// </summary>
     /// <param name="renderMode">The <see cref="IComponentRenderMode"/>.</param>
-    public void AddComponentRenderMode(IComponentRenderMode renderMode)
+    public void AddComponentRenderMode(IComponentRenderMode? renderMode)
     {
-        ArgumentNullException.ThrowIfNull(renderMode);
```

Comment:
I think you need to `return` explicitly if it's null, so it truly becomes a no-op. Otherwise you're setting `HasCallerSpecifiedRenderMode` even though there isn't a nonnull mode, and that will produce an actually different behavior than we had before.